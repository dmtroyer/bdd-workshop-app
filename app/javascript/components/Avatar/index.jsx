import React from 'react';
import PropTypes from 'prop-types';

const Avatar = ({handle}) => {
  return (
    <>
    <i className="fa fa-user" aria-hidden="true" alt="placeholder-avatar"></i>
    {handle ? (<span>{handle}</span>) : ''}
    </>
  )
};
Avatar.propTypes = {
  handle: PropTypes.string,
}

export default Avatar;
