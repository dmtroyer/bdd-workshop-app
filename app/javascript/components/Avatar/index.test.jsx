import React from 'react';
import {shallow} from 'enzyme';
import Avatar from '.';

jest.mock('../API');

describe('Avatar', () => {
  it('renders an avatar', () => {
    const wrapper = shallow(<Avatar/>);
    expect(wrapper).toMatchSnapshot()
  })
});