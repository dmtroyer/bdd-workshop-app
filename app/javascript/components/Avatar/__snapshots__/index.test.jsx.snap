// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders an avatar 1`] = `
<Fragment>
  <i
    alt="placeholder-avatar"
    aria-hidden="true"
    className="fa fa-user"
  />
</Fragment>
`;
